srv/salt/_modules/wait.py

```diff
@@ -144,6 +144,36 @@ def just(self):
         time.sleep(self.settings['delay'])
 
 
+class OSDInCheck(HealthCheck):
+    """
+    Check that all OSDs are "in"
+    """
+
+    def __init__(self, **kwargs):
+        kwargs.update({'nohealthcheck': True})
+        super(OSDInCheck, self).__init__(**kwargs)
+
+    def wait(self):
+        """
+        Poll until the status "matches" the specificed number of checks.
+        """
+        cmd = json.dumps({"prefix": "osd dump", "format": "json"})
+
+        def success(dump):
+            if 'osds' in dump:
+                for entry in dump['osds']:
+                    if 'in' in entry:
+                        if entry['in'] == 0:
+                            return False
+                    else:
+                        log.error("entry {} missing 'in' key".format(pprint.pformat(entry)))
+                        return False
+                return True
+            return False
```

Comment:
Minor nit, but perhaps this is slightly clearer with less `if` branches ..

```suggestion
        def success(dump):
            for entry in dump.get('osds', []):
                if not 'in' in entry:
                    log.error("entry {} missing 'in' key".format(pprint.pformat(entry)))
                    return False
                if entry.get('in', 0) != 1:
                    return False
            return True
```

Reply:
I'll take a look here... I wasn't thrilled with the number of returns either.  However, the issue with `dump.get('osds', [])` returning an empty list will result in success returning True.